docs: add known issues page (#1066)
neo4j-contrib/neo4j-spark-connector: docs/modules/ROOT/pages/faq.adoc

[#faq]
= FAQ and troubleshooting

[TIP]
====

> For the current limitations of the connector, see xref:known-issues.adoc[], which is the canonical list of known issues and workarounds.

Reviewer: I don't think we need to cross reference known limitations and faq entries to each other.

Reviewer 2: Agree I think having them next to each other in the left hand side bar is enoguh

=== Why are the returned columns not in the same order as I specified in the query?

> This is a known issue with Neo4j 3.x and Neo4j 4.0.

Reviewer: I have removed this section on my PR #1065.

In general, the solution is to repartition the DataFrame before writing it to Neo4j, to avoid
multiple partitioned writes from locking the same nodes and relationships.

> See xref:known-issues.adoc#deadlocks[Concurrent relationship writes can cause deadlocks].

Reviewer: Same, several points added around this topic in PR #1065.

all the values for that field are cast to String. This happens because Spark is not schema-free
and needs each column always to have its own type.

You can read more xref:read/schema.adoc[here], or see xref:known-issues.adoc#mixed-types[Cast errors when a property has different types across nodes].

=== I am getting an error _TableProvider implementation org.neo4j.spark.DataSource cannot be written with ErrorIfExists mode, please use Append or Overwrite modes instead_. What should I do?

> The connector does not support `SaveMode.ErrorIfExists`, which is the default save mode.

Reviewer: I think this is no longer true, and I removed this in #1065.

Reviewer 2: Indeed this is confusing, frankly I'm not 100% sure what we're supposed to support. But if you submit with "errorifexists" mode from pyspark you indeed get our validation error:

```
AnalysisException: [UNSUPPORTED_DATA_SOURCE_SAVE_MODE]
The data source "org.neo4j.spark.DataSource" cannot be written in the "ErrorIfExists" mode.
Please use either the "Append" or "Overwrite" mode instead. SQLSTATE: 0A000
```

Reviewer 2: I think what we're "supposed" to explicitly disallow (looking at legacy code and existing tests as reference) is ErrorIfExist on the relationship source/target save modes.

all the values for that field are cast to String. This happens because Spark is not schema-free
and needs each column always to have its own type.

> You can read more xref:read/schema.adoc[here], or see xref:known-issues.adoc#mixed-types[Cast errors when a property has different types across nodes].

Reviewer 3: ```suggestion
This is a xref:known-issues.adoc#mixed-types[known issue].
```
